src/wp-includes/block-template-utils.php

```diff
@@ -360,6 +360,17 @@ function _get_block_templates_files( $template_type, $query = array() ) {
 				continue;
 			}
 
+			// The child theme items (stylesheet) are processed before the parent theme's (template).
+			// If a child theme defines a template, prevent the parent template from being added to the list as well.
+			$slug_already_present = false !== array_search(
```

Comment:
This logic was missing: https://github.com/WordPress/wordpress-develop/pull/4097/files#r1279431292